specs/safe-liveness-checking.md

```diff
@@ -0,0 +1,66 @@
+# Safe Liveness Checking
+
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+**Table of Contents**
+
+- [Overview](#overview)
+- [Liveness Checking Mechanism](#liveness-checking-mechanism)
+- [Liveness Check Implementation](#liveness-check-implementation)
+- [Liveness Check Validation](#liveness-check-validation)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+
+## Liveness Checking Mechanism
+
+The Security Security Council uses a specially extended multisig...
+
+
+## Liveness checking methodology
+
+This is achieved using two types of contracts which the Safe contract has built-in support for:
+
+1. **Guard contracts:** can execute pre- and post- transaction checks.
+1. **Module contracts:** a contract which is added to the Safe by the signers, and thenceforth is
+   authorized to execute transactions via the Safe. This means the module must properly implement
+   auth conditions internally.
+
+### The Liveness Guard
+
+For implementing liveness checks a `LivenessGuard` is created which receives the signatures from
+each executed transaction, and tracks the latest time at which a transaction was signed by each
+signer. This time is made publicly available by calling a `lastSigned(address)(Timestamp)` method.
+
+Signers may also call the contract directly in order to prove liveness.
```

Comment:
So the `LivenessGuard` only works for executed transactions, but transactions that are signed and don't meet the execution threshold get ignored, since they never land on-chain. This is arguably ok since if a signer is genuinely affected by this, they can just call the contract to prove liveness.

Though, a disinterested signer can repeatedly call `showLiveness` to stay live without actually contributing anything. Maybe worth seeing if all signatures can be managed on-chain instead of relying on Safe's servers to store sigs? Then you can remove the liveness check and only rely on signatures

Reply:
Good points.

Regarding unexecuted but signed tx, I agree that's annoying for signers, but should be sufficiently mitigated by `showLiveness`. 

WRT a 'disinterested signer' (I assume you mean one that's not actively participating), I think it's OK. We should keep this guard/module pure in that they only enforce liveness checking. Active participation is a more subjective consideration, and it should be up to governance vote for, and the other signers to enact, the removal of an inactive signer.